### PR TITLE
[gcp][feat] Add firestore service collection

### PR DESCRIPTION
**Pull request by 1101-1**
# Description

# To-Dos

- [x] I have created tests for any new or updated functionality.
- [x] I ran `tox` successfully.

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://inventory.fix.security/code-of-conduct).

### CODE REVIEW

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py [hunk 1]
```diff
@@ -2597,19 +2571,6 @@ class GcpVertexAIPipelineTaskDetail:
...
-        service="aiplatform",
-        version="v1",
-        service_with_region_prefix=True,
-        accessors=["projects", "locations", "metadataStores", "contexts"],
-        action="list",
-        request_parameter={"parent": "projects/{project}/locations/{region}"},
-        request_parameter_in={"project", "location"},
-        response_path="contexts",
-        response_regional_sub_path=None,
-    )
```
Comment (aquamatthias) [inline plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:2612]: any reason?

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py [hunk 2]
```diff
@@ -1886,19 +1886,6 @@ def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
...
-        service="aiplatform",
-        version="v1",
-        service_with_region_prefix=True,
-        accessors=["projects", "locations", "metadataStores", "artifacts"],
-        action="list",
-        request_parameter={"parent": "projects/{project}/locations/{region}"},
-        request_parameter_in={"project", "region"},
-        response_path="artifacts",
-        response_regional_sub_path=None,
-    )
```
Comment (aquamatthias) [inline plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:1901]: any reason?

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py [hunk 3]
```diff
@@ -2519,19 +2506,6 @@ class GcpVertexAIPipelineTaskDetailPipelineTaskStatus:
...
-        service="aiplatform",
-        version="v1",
-        service_with_region_prefix=True,
-        accessors=["projects", "locations", "metadataStores", "executions"],
-        action="list",
-        request_parameter={"parent": "projects/{project}/locations/{region}"},
-        request_parameter_in={"project", "location"},
-        response_path="executions",
-        response_regional_sub_path=None,
-    )
```
Comment (aquamatthias) [inline plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:2534]: any reason?

**Review by aquamatthias — COMMENTED** [review]
one question about deleted lines in aiplatform

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py (hunk 1 above)
Comment (1101-1) [reply plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:2612]: `GcpVertexAIContext` used as subresource
```python3

@define(eq=False, slots=False)
class GcpVertexAIPipelineJobDetail:
    kind: ClassVar[str] = "gcp_vertex_ai_pipeline_job_detail"
    mapping: ClassVar[Dict[str, Bender]] = {
        "pipeline_context": S("pipelineContext", default={}) >> Bend(GcpVertexAIContext.mapping),
        "pipeline_run_context": S("pipelineRunContext", default={}) >> Bend(GcpVertexAIContext.mapping),
```

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py (hunk 3 above)
Comment (1101-1) [reply plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:2534]: `GcpVertexAIExecution` used as subresource:
```python3
@define(eq=False, slots=False)
class GcpVertexAIPipelineTaskDetail:
    kind: ClassVar[str] = "gcp_vertex_ai_pipeline_task_detail"
    mapping: ClassVar[Dict[str, Bender]] = {
        "create_time": S("createTime"),
        "end_time": S("endTime"),
        "rpc_error": S("error", default={}) >> Bend(GcpGoogleRpcStatus.mapping),
        "execution": S("execution", default={}) >> Bend(GcpVertexAIExecution.mapping),
 ```

File: plugins/gcp/fix_plugin_gcp/resources/aiplatform.py (hunk 2 above)
Comment (1101-1) [reply plugins/gcp/fix_plugin_gcp/resources/aiplatform.py:1901]: `GcpVertexAIArtifact` used as subresource:
```python3
class GcpVertexAIPipelineTaskDetailArtifactList:
    kind: ClassVar[str] = "gcp_vertex_ai_pipeline_task_detail_artifact_list"
    mapping: ClassVar[Dict[str, Bender]] = {
        "artifacts": S("artifacts", default=[]) >> ForallBend(GcpVertexAIArtifact.mapping)
    }
    artifacts: Optional[List[GcpVertexAIArtifact]] = field(default=None)
```

**Review by aquamatthias — APPROVED** [review]

**Pull request merged**